test/e2e/util.go

```diff
@@ -282,3 +282,31 @@ func addressIsIP(address v1.NodeAddress) bool {
 	}
 	return true
 }
+
+// Returns pod's ipv4 and ipv6 addresses IN ORDER
+func getPodAddresses(pod *v1.Pod) (string, string) {
+	var ipv4Res, ipv6Res string
+	for _, a := range pod.Status.PodIPs {
+		if utilnet.IsIPv4String(a.IP) {
+			ipv4Res = a.IP
+		}
+		if utilnet.IsIPv6String(a.IP) {
+			ipv6Res = a.IP
+		}
+	}
+	return ipv4Res, ipv6Res
+}
+
+// Returns nodes's ipv4 and ipv6 addresses IN ORDER
+func getNodeAddresses(node *v1.Node) (string, string) {
```

Comment:
just only FYI, there are a bunch of these here https://github.com/kubernetes/kubernetes/blob/23a46d88439e5919fa61d5b25e5752c258a900fb/test/e2e/framework/node/resource.go#L283-L292